packages/interactivity/src/directives.js

```diff
@@ -205,21 +205,28 @@ const cssStringToObject = ( val ) => {
  * @param {string} type 'window' or 'document'
  * @return {void}
  */
-const getGlobalEventDirective =
-	( type ) =>
-	( { directives, evaluate } ) => {
+const getGlobalEventDirective = ( type ) => {
+	return ( { directives, evaluate } ) => {
+		const events = new Map();
 		directives[ `on-${ type }` ]
 			.filter( ( { suffix } ) => suffix !== 'default' )
 			.forEach( ( entry ) => {
+				const event = entry.suffix.split( '--' )[ 0 ];
+				if ( ! events.has( event ) ) events.set( event, new Set() );
+				events.get( event ).add( entry );
+			} );
+		events.forEach( ( entries, eventType ) => {
```

Comment:
What's the motivation for doing this in two passes (iterate and collect in Map, then iterate on the map) instead of setting the event listeners in a single pass? I see this is the approach used in #60661 as well.

Reply:
I did the same approach as #60661 to keep consistency in the codebase. Do you want me to refactor both on this PR or make a follow-up one?

Reply:
If the collection and second iteration is redundant, it seems preferable to do it on a single pass.

> Do you want me to refactor both on this PR or make a follow-up one?

In this PR is fine with me, but I'll leave it up to whatever you prefer.

Reply:
It turns out this PR is different from the other PR. Here, we're using `addEventListener` to add listeners to `window` or `document`, but that other PR is adding multiple event listeners effectively via props. That PR therefor creates a single listener that needs to call all of the listeners internally and passes it via the `on${Event}` prop:

https://github.com/WordPress/gutenberg/blob/cbdf5406c57d822b8339e030bd16b3783cd7b061/packages/interactivity/src/directives.js#L283-L287

We don't have that same constraint here, each event listener can be added and removed on its own, so I think we'll need to keep the approach used in the other PR, while this PR can remove the collection + iteration and add the listeners in a single pass.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/61009/commits/d57523d436c0283ecde04c02a3ceade833edb682